### PR TITLE
8068824: Exception thrown in JTableHeader after clicking on popupmenu opened with right-click on header

### PR DESCRIPTION
**Pull request by prsadhuk**
The issue is observed on mouse right click on the table header to open a pupup menu and then, do a mouse left click on the menu entry in which case we get AIOOBE
This is because the columnIndex for popup menu is returned by `viewIndexForColumn `as -1 seen here https://github.com/openjdk/jdk/blob/941a7ac7dab243c6033a78880fd31faa803e62ab/src/java.desktop/share/classes/javax/swing/plaf/basic/BasicTableHeaderUI.java#L745-L753
which is then passed to DefaultTableColumnModel.getColumn(columnIndex) causing it to throw AIOOBE as per the [DefaultTableColumnModel.getColumn spec](https://github.com/openjdk/jdk/blob/master/src/java.desktop/share/classes/javax/swing/table/DefaultTableColumnModel.java#L293)

So, inline with check done for `viewIndexForColumn `
https://github.com/openjdk/jdk/blob/941a7ac7dab243c6033a78880fd31faa803e62ab/src/java.desktop/share/classes/javax/swing/plaf/basic/BasicTableHeaderUI.java#L310-L312
a similar check is added when `viewIndexForColumn `is called

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8068824](https://bugs.openjdk.org/browse/JDK-8068824): Exception thrown in JTableHeader after clicking on popupmenu opened with right-click on header

### Reviewers
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Committer) ⚠️ Review applies to [130baf1a](https://git.openjdk.org/jdk/pull/13172/files/130baf1a3b4a4a0b029aba78fb379ea91dc7d872)
 * [Ajit Ghaisas](https://openjdk.org/census#aghaisas) (@aghaisas - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13172/head:pull/13172` \
`$ git checkout pull/13172`

Update a local copy of the PR: \
`$ git checkout pull/13172` \
`$ git pull https://git.openjdk.org/jdk.git pull/13172/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13172`

View PR using the GUI difftool: \
`$ git pr show -t 13172`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13172.diff">https://git.openjdk.org/jdk/pull/13172.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13172#issuecomment-1482596710)

### CODE REVIEW

File: test/jdk/javax/swing/JTableHeader/JTableHeaderExceptionTest.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+ * @summary JTable header rendering problem (after setting preferred size)
+ * @run JTableHeaderExceptionTest
+*/
+import java.awt.BorderLayout;
+import java.awt.MouseInfo;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.ActionEvent;
+import java.awt.event.InputEvent;
+import javax.swing.*;
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/JTableHeader/JTableHeaderExceptionTest.java:37]: Wild import can be removed.

File: test/jdk/javax/swing/JTableHeader/JTableHeaderExceptionTest.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        } finally {
+            SwingUtilities.invokeAndWait(frame::dispose);
+        }
+    }
+}
+
+class DragTestTable extends JTable {
+
+    Point loc;
+    public DragTestTable(DragTestTableModel model){
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/JTableHeader/JTableHeaderExceptionTest.java:91]: Please add a space here `public DragTestTable(DragTestTableModel model) {`

File: test/jdk/javax/swing/JTableHeader/JTableHeaderExceptionTest.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+        super(model);
+
+        JPopupMenu menu = new JPopupMenu();
+        menu.add(new AbstractAction("Bug Test") {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                //loc = menu.getLocationOnScreen();
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/JTableHeader/JTableHeaderExceptionTest.java:99]: Commented line can be removed.

File: test/jdk/javax/swing/JTableHeader/JTableHeaderExceptionTest.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+
+        getTableHeader().setComponentPopupMenu(menu);
+    }
+
+    public static class DragTestTableModel extends AbstractTableModel {
+        private Object rowData[][];
+        private Object columnNames[];
+
+        public DragTestTableModel(Object rowData[][], Object columnNames[]){
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/JTableHeader/JTableHeaderExceptionTest.java:113]: Please add a space here `public DragTestTableModel(Object rowData[][], Object columnNames[]) {`

**Comment by kumarabhi006** [comment]
Please split the long line and update the copyright year.

**Review by kumarabhi006 — APPROVED** [review]
Test verified. Looks good to me.

**Review by aghaisas — APPROVED** [review]
Looks good to me!

**Pull request closed without merging**